tensorflow/core/util/tensor_bundle/tensor_bundle_test.cc

```diff
@@ -259,6 +272,207 @@ void TestBasic() {
   }
 }
 
+// Reinterpret the bytes of an rvalue
+#define _CAST(x, T) (*reinterpret_cast<const T*>(&x))
+
+// Type-specific subroutine of SwapBytes test below
+template <typename T>
+void TestByteSwap(const T* forward, const T* swapped, int array_len) {
+  auto bytes_per_elem = sizeof(T);
+
+  // Convert the entire array at once
+  std::unique_ptr<T> forward_copy(new T[array_len]);
+  std::memcpy(forward_copy.get(), forward, array_len * bytes_per_elem);
+  TF_EXPECT_OK(
+      ByteSwapArray((char*)forward_copy.get(), bytes_per_elem, array_len));
+  for (int i = 0; i < array_len; i++) {
+    EXPECT_EQ(forward_copy.get()[i], swapped[i]);
+  }
+
+  // Then the array wrapped in a tensor
+  auto shape = TensorShape({array_len});
+  auto dtype = DataTypeToEnum<T>::value;
+  Tensor forward_tensor(dtype, shape);
+  Tensor swapped_tensor(dtype, shape);
+  std::memcpy((char*)forward_tensor.tensor_data().data(), forward,
```

Comment:
C-style casts are discouraged; use const_cast

Reply:
Switched to const_cast